lib/cloudant.js

```diff
@@ -75,7 +75,42 @@ Cloudant.prototype.getTypes = function() {
  */
 Cloudant.prototype.connect = function(cb) {
   debug('Cloudant.prototype.connect');
-  cb(null, this.cloudant);
+  var self = this;
+  // check if cloudant server is up
+  if (self.options.url) {
+    request.get(self.options.url, function(err, res, body) {
```

Comment:
I removed the ping code in the previous commit because calling the driver in the callback of the `request.get` would give us the same issue as using the async version of our driver (which is not getting the cloudant connection object back in time when we use it further down the line).